crates/driver/src/infra/blockchain/mod.rs

```diff
@@ -18,8 +17,11 @@ pub struct Ethereum {
 impl Ethereum {
     /// Access the Ethereum blockchain through an RPC API hosted at the given
     /// URL.
-    pub async fn ethrpc(url: &Url) -> Result<Self, web3::Error> {
-        // TODO Probably move shared::ethrpc into its own crate and reuse it here
+    pub async fn ethrpc(url: &url::Url) -> Result<Self, web3::Error> {
+        // TODO Enable batching, reuse ethrpc? Put it in the boundary module?
+        // I feel like what we have in shared::ethrpc could be simplified if we use
+        // web3::transports::batch or something, but I haven't looked deep into it, just
```

Comment:
We use `web3::transports::batch` under the hood. The issue is that it only allows manual batching AFAIK.